lib/zendesk_apps_support/validations/source.rb

```diff
@@ -1,68 +1,23 @@
 # frozen_string_literal: true
 
-require 'jshintrb'
-
 module ZendeskAppsSupport
   module Validations
     module Source
-      LINTER_OPTIONS = {
-        # enforcing options:
-        noarg: true,
-        undef: true,
-
-        # relaxing options:
-        eqnull: true,
-        laxcomma: true,
-        sub: true,
-
-        # predefined globals:
-        predef: %w[_ console services helpers alert confirm self
-                   JSON Base64 require module exports moment],
-
-        browser: true
-      }.freeze
-
-      class <<self
+      class << self
         def call(package)
-          files = package.js_files
-          app   = files.find { |file| file.relative_path == 'app.js' }
-
-          if package_needs_app_js?(package)
-            return [ ValidationError.new(:missing_source) ] unless app
-          else
-            return (package_has_code?(package) ? [ ValidationError.new(:no_code_for_ifo_notemplate) ] : [])
+          if no_source_required_app?(package.manifest) && contain_source_files?(package)
+            ValidationError.new(:no_code_for_ifo_notemplate)
           end
-
-          jshint_errors(files).flatten!
         end
 
         private
 
-        def package_has_code?(package)
-          !(package.js_files.empty? && package.template_files.empty? && package.app_css.empty?)
-        end
-
-        def package_needs_app_js?(package)
-          return false if package.manifest.marketing_only?
-          return false if package.manifest.requirements_only?
-          return false if package.manifest.iframe_only?
-          true
-        end
-
-        def jshint_error(file)
-          errors = linter.lint(file.read)
-          [JSHintValidationError.new(file.relative_path, errors)] if errors.any?
-        end
-
-        def jshint_errors(files)
-          files.each_with_object([]) do |file, errors|
-            error = jshint_error(file)
-            errors << error unless error.nil?
-          end
+        def contain_source_files?(package)
+          package.js_files.any? || package.template_files.any? || !package.app_css.empty?
         end
 
-        def linter
-          Jshintrb::Lint.new(LINTER_OPTIONS)
+        def no_source_required_app?(manifest)
+          manifest.requirements_only? || manifest.marketing_only?
```

Comment:
What about `.iframe_only?` - do we still need that here?

Reply:
The definition of`.iframe_only?` here refers to any [V2 app](https://github.com/zendesk/zendesk_apps_support/blob/master/lib/zendesk_apps_support/manifest.rb#L106).

Any V2 app would typically have source files such as HTML, CSS and JS. Therefore it wouldn't be classified under `no_source_required_app?`

Putting them together, we certainly do not want to block app upload with a validation because they package.manifest.is_a_v2_app && contain_source_files?(package).

Therefore I do not think we still need that there. What do you think?

Reply:
I'm pretty sure an iframe-only V2 app can just link out to some externally-hosted assets in the manifest, rather than having CSS/JS in the zip itself.

Reply:
Oh yes I agree Sean, a V2 app can have an internal source, ie. `assets/ticket_sidebar.js` and `assets/ticket_sidebar.css` AND a V2 app can have a location pointing to a server hosted somewhere else.

Hence I have removed this check in the PR. We do not want to block app upload with any validation error if the app is of both types. Let's discuss this further in person :)

Reply:
As discussed in person, I will rename this method to be `app_doesnt_require_source?` to avoid confusion. Thank you for the suggestion!